emscripten/CMakeLists.txt

```diff
@@ -0,0 +1,20 @@
+add_executable(deqp.js deqp.js.cpp ${DEQP_MODULE_ENTRY_POINTS})
+target_link_libraries(deqp.js tcutil-platform ${DEQP_MODULE_LIBRARIES})
+add_custom_target(deqp.js-data)
+add_dependencies(deqp.js deqp.js-data)
+set_target_properties(deqp.js PROPERTIES
+	OUTPUT_NAME "deqp"
+	SUFFIX ".js"
+	LINK_FLAGS "--bind")
+# TODO(kainino@chromium.org): These should happen programmatically.
+add_data_dir(deqp.js ../data/gles2/data gles2/data)
+add_data_dir(deqp.js ../data/gles2/shaders gles2/shaders)
+add_data_dir(deqp.js ../data/gles3/data gles3/data)
+add_data_dir(deqp.js ../data/gles3/shaders gles3/shaders)
+
+add_custom_command(
+	TARGET deqp.js
+	POST_BUILD
+	COMMAND ${CMAKE_COMMAND} -E copy
+		${CMAKE_CURRENT_SOURCE_DIR}/deqp.html
+		${CMAKE_CURRENT_BINARY_DIR}/deqp.html)
```

Comment:
I would favor setting the build output to `.html` and using Emscripten's `--shell-file` parameter to do this, rather than building `.js` and managing everything yourself. That's because there are a couple of features that Emscripten customizes the html file (although now that I think about it, perhaps if we just target wasm, those are never used). What's the motivation of `-o deqp.js` instead of `-o deqp.html`?

Reply:
This deqp.html is really just a demo of the JS library; ultimately I'll want to find a way to make this runnable as part of the WebGL conformance test suite, or at least as part of other test harnesses (e.g. in Chrome).

So I don't want a main function that runs immediately and runs all tests, or even one that runs immediately and runs a particular subset of tests. At least for flexibility, I want a single instance of the page to be able to do multiple test runs.

Reply:
Right now since we don't have caching of the wasm compiled code, this makes it possible to do multiple test runs without waiting 4+ seconds (more with ES 3) between each run.